presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -391,8 +391,12 @@ public LocalExecutionPlan plan(
         }
         else {
             partitionChannels = partitioningScheme.getPartitioning().getArguments().stream()
-                    .map(ArgumentBinding::getColumn)
-                    .map(outputLayout::indexOf)
+                    .map(argument -> {
+                        if (argument.isConstant()) {
+                            return -1;
```

Comment:
This logic produces the same result as before except it's explicit about handling constant differently from handling variables. Since `PartitionedOutputOperator` expects `partitionChannels` and `partitionConstants` has the same length as the number of arguments, we need to designate a special value (position holder) for constants in `partitionChannels`. `PartitionedOutputOperator.getPartitionFunctionArguments` would not try to use this channel if it's a constant. I don't like using -1 as a place holder. If we are willing to change `partitionChannels` to List<Optional<Integer>> it might be better. Not sure it's worth it though.